src/modules/AddOnProduct.rb

```diff
@@ -885,9 +886,15 @@ def RegisterAddOnProduct(src_id)
       # or check the content file
       if WorkflowManager.WorkflowRequiresRegistration(src_id) || Builtins.contains(@addons_requesting_registration, src_id)
         Builtins.y2milestone("Repository ID %1 requests registration", src_id)
-        # TODO FIXME: user needs to manually select the addon to register,
-        # pass the addon so it could be pre-selected
-        WFM.CallFunction("inst_scc", [])
+
+        if !WFM.ClientExists("inst_scc") && !Package.Install("yast2-registration")
```

Comment:
Having something like the installation of a package kinda "hidden" as the second part of a condition looks strange to me.